### PR TITLE
configobserver: move install-config CIDR readers here

### PR DESCRIPTION
**Pull request by sttts**
This code is used by multiple operators.

### CODE REVIEW

File: pkg/operator/configobserver/network/observe_network.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"github.com/openshift/library-go/pkg/operator/events"
```
Comment (mfojtik) [inline pkg/operator/configobserver/network/observe_network.go]: nit: imports

File: pkg/operator/configobserver/network/observe_networking_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
+package network
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/openshift/library-go/pkg/operator/events"
```
Comment (mfojtik) [inline pkg/operator/configobserver/network/observe_networking_test.go]: nit: move this down

File: pkg/operator/configobserver/network/observe_network.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	"k8s.io/client-go/listers/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
```
Comment (mfojtik) [inline pkg/operator/configobserver/network/observe_network.go]: @sttts @deads2k we hardcode this in so many places, I almost wish we have some const package in library-go to control this from one place?

File: pkg/operator/configobserver/network/observe_network.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	clusterConfigNamespace = "kube-system"
+	clusterConfigName      = "cluster-config-v1"
+)
+
+// GetClusterCIDRs reads the cluster CIDRs from the install-config ConfigMap in the cluster.
+func GetClusterCIDRs(lister v1.ConfigMapLister, recorder events.Recorder) ([]string, error) {
+	clusterConfig, err := lister.ConfigMaps(clusterConfigNamespace).Get(clusterConfigName)
+	if errors.IsNotFound(err) {
+		recorder.Warningf("ObserveClusterCIDRFailed", "Required %s/%s config map not found", clusterConfigNamespace, clusterConfigName)
+		glog.Warning("configmap/cluster-config-v1.kube-system: not found")
```
Comment (mfojtik) [inline pkg/operator/configobserver/network/observe_network.go]: @deads2k @sttts every event also generate a glog message when created... I wonder if we can deduplicate these to get cleaner logs?

File: pkg/operator/configobserver/network/observe_network.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/configobserver/network/observe_network.go]: fixed

File: pkg/operator/configobserver/network/observe_network.go (hunk 4 above)
Comment (sttts) [reply pkg/operator/configobserver/network/observe_network.go]: a configobserver/log package. Same for events.

**Comment by mfojtik** [comment]
/lgtm

follow-up 1: cleanup the logging/event reporting code
follow-up 2: have some pkg/consts package with common namespaces/names?

File: pkg/operator/configobserver/network/observe_network.go (hunk 3 above)
Comment (deads2k) [reply pkg/operator/configobserver/network/observe_network.go]: > @sttts @deads2k we hardcode this in so many places, I almost wish we have some const package in library-go to control this from one place?

no const package

File: pkg/operator/configobserver/network/observe_network.go (hunk 4 above)
Comment (deads2k) [reply pkg/operator/configobserver/network/observe_network.go]: > @deads2k @sttts every event also generate a glog message when created... I wonder if we can deduplicate these to get cleaner logs?

you mean just delete the glog?

**Comment by deads2k** [comment]
> follow-up 2: have some pkg/consts package with common namespaces/names?

No.  We should not do that.  People own their own location.  For example, we know these are moving.

**Pull request merged**